tests/run-pass/fs.rs

```diff
@@ -42,6 +42,15 @@ fn main() {
     file.read_to_end(&mut contents).unwrap();
     assert_eq!(bytes, contents.as_slice());
 
+    // Cloning a file should be successful.
+    let file = File::open(&path).unwrap();
+    let mut cloned = file.try_clone().unwrap();
+    // Reading from a cloned file should get the same text.
+    let mut contents = Vec::new();
+    cloned.read_to_end(&mut contents).unwrap();
+    assert_eq!(bytes, contents.as_slice());
```

Comment:
Did I mention that this is turning into a monster function?^^

Would be better to split it up before we let it grow further.

Reply:
#1152 splits this file into multiple functions, are you okay with waiting for that to land and then rebasing?

Reply:
Well, #1152 is the most complex of your PRs so it is likely going to take longest to review.